### PR TITLE
Issue #2687 Modify tray started check to verify if daemon is running

### PR DESCRIPTION
**Pull request by anjannath**
the daemon might take a few seconds to come up and make the api endpoint available
it is better to check that the daemon is up by trying to access this endpoint  to
make sure user is successfully able to do `crc start` after `crc setup`

### CODE REVIEW

File: pkg/crc/preflight/preflight_checks_tray_darwin.go [hunk 1]
```diff
@@ -99,3 +103,12 @@ func fixPlistFileExists(agentConfig launchd.AgentConfig) error {
 	}
 	return nil
 }
+
+func checkIfDaemonRunning() error {
+	return crcerrors.Retry(context.Background(), time.Second*20, func() error {
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_tray_darwin.go:108]: 20 seconds seem very long?

Comment (anjannath) [reply pkg/crc/preflight/preflight_checks_tray_darwin.go:108]: yes, it needs to be changed, in the on macOS tray we only wait 1 second before making `/status` calls, so 1 second interval and the limit we can set to 3? worst case 3 seconds wait time

File: pkg/crc/preflight/preflight_checks_tray_darwin.go [hunk 2]
```diff
@@ -99,3 +103,12 @@ func fixPlistFileExists(agentConfig launchd.AgentConfig) error {
...
 	return nil
 }
+
+func checkIfDaemonRunning() error {
+	return crcerrors.Retry(context.Background(), time.Second*3, func() error {
+		if _, err := daemonclient.New().APIClient.Version(); err != nil {
+			return &crcerrors.RetriableError{Err: fmt.Errorf("Daemon is not yet running: %w", err)}
+		}
+		return nil
+	}, time.Second*1)
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_tray_darwin.go:113]: the `*1` is not needed :)

**Review by cfergeau — APPROVED** [review]

**Comment by praveenkumar** [comment]
Why this is marked as `wip` ?

**Comment by anjannath** [comment]
> Why this is marked as wip ?

Its removed now, the bot took sometime to remove the label :)

**Pull request merged**